packages/server/graphql/mutations/helpers/isStartMeetingLocked.ts

```diff
@@ -1,12 +1,30 @@
+import isValid from '../../../graphql/isValid'
+import getTeamsByOrgIds from '../../../postgres/queries/getTeamsByOrgIds'
 import {DataLoaderWorker} from '../../graphql'
 
 const isStartMeetingLocked = async (teamId: string, dataLoader: DataLoaderWorker) => {
   const team = await dataLoader.get('teams').loadNonNull(teamId)
-  const {isPaid, lockMessageHTML} = team
-  if (isPaid) return null
-  return lockMessageHTML
-    ? 'Wow, you’re determined to use Parabol! That’s awesome! Do you want to keep sneaking over the gate, or walk through the door with our Sales team?'
-    : 'Sorry! We are unable to start your meeting because your team has an overdue payment'
+  const {lockMessageHTML, orgId, tier} = team
+  const orgTeams = await getTeamsByOrgIds([orgId])
+  const anyOrgTeamIsLocked = orgTeams.some((team) => !team.isPaid)
```

Comment:
I think we should check whether the team is part of an org that hasn't paid. Otherwise, if we just check the team and they create a new team within the same org, they'll be able to use that team as its `tier` will be pro/enterprise. 

Also, why do we lock by team rather than org when the org is the billing entity?

Reply:
-1 `isPaid` is denormalized onto the team so if 1 is paid, they're all paid.
100% this should live on the organization, it should be refactored...

all that to say we can avoid that extra DB hit on L8 if tier !== personal

https://github.com/ParabolInc/parabol/blob/4172e4446fb0b6206fa585a36d037476899e6330/packages/server/billing/helpers/terminateSubscription.ts

Reply:
> if 1 is paid, they're all paid

When an org pays, all teams in that org are initially paid. However, when we lock a team, we only update `isPaid` on that team ([see here](https://github.com/ParabolInc/parabol/blob/234bc53534684b29b70e2a4357ba27f42bd1fa20/packages/server/graphql/private/mutations/lockTeams.ts#L16)), so one team could be paid while another isn't. If we don't check whether other teams in the pro/enterprise org are paid, they could keep creating new teams in the org. 

One concern with this approach is if an org has three teams but says they'll never pay, we'd lock the third team so they remain within the two-team free tier. However, with this implementation, they/we would need to delete the third team so that their other two teams can still create meetings. I think that's OK - I'd rather force a conversation with sales than be too lenient.

Reply:
ah, i gotcha.
@gcrickman when you lock a team with a message is it OK if we lock every team on that org? I forget why we lock teams instead of orgs...

Reply:
[This is why: ](https://github.com/ParabolInc/parabol/issues/6378#issuecomment-1190556784)
> A few things can happen:
> - A team lead may choose to pay for their own teams, but then others within the org need to be locked out
> - A company may say 'oh we'll just fit into the free tier' - then we'll lock all but 2 teams, since they can have 2 teams on the free tier

Reply:
> we'd need to delete the teams the free-tier org no longer wants to use. Would that work for you?

Can you say more about this? So instead of locking the teams, we'd delete the teams? 

The reason we want to lock is: 
- We want folks trying to get access to get a message about what's happening, so they could request to unlock or trigger a conversation with their own leadership about paying 
- If they do move to paid, they can regain access to their history and data in Parabol

Reply:
> The reason we want to lock is:
> - We want folks trying to get access to get a message about what's happening, so they could request to unlock or trigger a conversation with their own leadership about paying
> - If they do move to paid, they can regain access to their history and data in Parabol

This would still be possible. Let's say Cheeky Org has three teams: Team A, Team B, and Team C. They're on the free-tier and don't want to pay, so we lock Team C. Clicking on Team C no longer shows their tasks - it shows the lock modal. Teams A & B can still see their tasks. All three teams can no longer start new meetings and see a helpful error message when they try.

If Cheeky Org does pay, we'll unlock Team C so they have access to their tasks and all teams can create meetings.

If they don't pay, we would archive Team C, so that Teams A & B can start meetings, which is easily done in GraphiQL.

Here's a Loom that demo's how it's currently working: https://www.loom.com/share/8e395fb8b8c94c7fa3ede02cbd12462f

Reply:
thanks for the reminder @avivapinchas! That makes a bunch of sense.

@nickoferrall I think your logic is sound. Ultimately, we need to move away from the team-based limit & limit on something that compels them to upgrade on their own, but that's another discussion.

Reply:
Thanks for that video, super helpful @nickoferrall! 

How I would want this to work is: 
- Teams A & B can still run their meetings
- Members of Team C cannot run meetings *and* cannot create new teams 

The reason for that is: 
- Teams A and B, whether paying or fitting into the free tier, are doing the right thing - we shouldn't put barriers in front of them 
- Team C should visibly see the problem - I don't want to archive them because then whatever attachment they have to their history or tasks or what have you goes away, they have less reason to come back. 

Is it possible to adjust this?

Reply:
Actually, I have a clarifying question: If the team members in Team A & B are _not_ members of Team C (there is no overlap in users), can they still run meetings? And: can a team member still leave Team C?

Reply:
> If the team members in Team A & B are not members of Team C (there is no overlap in users), can they still run meetings? And: can a team member still leave Team C?

Nope, the current implementation prevents everyone in Cheeky Org from starting meetings. The locked modal would cover the screen preventing users from leaving Team C, but they could leave the org.

The way you would like it to work makes sense to me, but I have one concern:
- catherine@cheeky.org from Team C sees that her team is locked and she can't create a new org/team
- Catherine creates a new user account with email catherine+stingy@cheeky.org
- She creates a new team or org, adds users from Team C, and starts a new meeting
- Alternatively, Catherine asks her colleague in Team A to create a new team/org

We could prevent members of Team C from creating a new team or org by:
- In the [addTeam](https://github.com/ParabolInc/parabol/blob/3939ec06398841ddb42e87c2b3a19683ac112370/packages/server/graphql/mutations/addTeam.ts) and [addOrg](https://github.com/ParabolInc/parabol/blob/3939ec06398841ddb42e87c2b3a19683ac112370/packages/server/graphql/mutations/addOrg.ts) mutations, check if the person creating the team belongs to any locked teams. If they do, show an error message.
- Remove the logic added in this PR to `isStartMeetingLocked`. As is the case in prod currently, members of Team C wouldn't be able to start a meeting, but members of Team A & B could. Members of any other org could too.

For this to be an effective restriction, I think we'd need to prevent users that belong to a locked team from joining new teams or orgs. We could do this by:
- Checking if the user belongs to a locked team in [acceptTeamInvitation](https://github.com/ParabolInc/parabol/blob/218c326c9f6c50f9d1496f2b25041a628bfa0a40/packages/server/safeMutations/acceptTeamInvitation.ts)
- If they do, show the locked modal and don't let them join

The concern here is a consultant could belong to a locked team, have no say in the billing, but need to join another org. They would need to remove themself from Cheeky Org to be able to do that.

----

I see two options:

- The current implementation which prevents teams from creating a new meeting if they have a team member that belongs to a locked org
- Prevent users from creating new teams/orgs and from joining teams if they belong to a locked team

What do you think @avivapinchas @mattkrick?

Reply:
I like option 2. 

> The current implementation which prevents teams from creating a new meeting if they have a team member that belongs to a locked team

So would members of Team A&B be able to run meetings, if they had no overlapping team members on Team C?

Reply:
i've got some tension with sprinkling in too much business logic around this use case, but as long as it doesn't turn into a cat & mouse game & stops here I think we're OK wrt ROI.

I'd argue we should only handle this logic in `accceptTeamInvitation` and NOT in creating new teams/orgs. That'll keep it simpler & teams are pointless if their friends can't join.

I'd add the logic in parallel to this call: https://github.com/ParabolInc/parabol/blob/7279d1fd87288b5629b993fc292ad978baaa9c0d/packages/server/graphql/public/mutations/acceptTeamInvitation.ts#L62

Reply:
So what would it loo like for locked users when they go to create new teams/orgs? They could do it, but then when they go to invite, they see the message?

Reply:
they could create a new team, they could invite new people, but those new people couldn't join. when they tried they would receive a snackbar notification saying something like "You're over limit, freeloader"

Reply:
🤔 I think it's clearer/stronger to show the notice to the person trying to skirt the rules, vs the folks they invited, buuuuttt maybe fun things will happen by doing it this way! Safe to try to move forward with this and see what the impact is.